### PR TITLE
Python version CI tests for `[3.6, 3.7, 3.8, 3.9]`

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Expanded Python version testing for YOLOv5 continuous integration.

### 📊 Key Changes
- Added multiple Python versions (3.6, 3.7, 3.8) to the CI testing matrix.
- Removed superfluous whitespace at the beginning of the CI testing file.

### 🎯 Purpose & Impact
- **Broader Compatibility:** Testing across various Python versions ensures wider compatibility and identifies any version-specific issues early.
- **CI Efficiency:** Clean-up of the CI configuration file for better readability and maintenance.
- **User Confidence:** Users can trust that YOLOv5 works with their Python environment, fostering a reliable experience across different setups. 🛠️

### CODE REVIEW

**Review by SpongeBab — DISMISSED** [review]

**Comment by WWE-Network** [comment]
https://github.com/WWE-Network/docs/tree/1862-Add-Travis-CI-migration-table

**Review by WWE-Network — DISMISSED** [review]

**Review by ian475 — APPROVED** [review]

**Review by ian475 — APPROVED** [review]
Thank you for review

**Review by ian475 — DISMISSED** [review]
bdbdca24794eaeefccf6cc4dd982ca92d7b0ccf6

**Comment by NauchtanRobotics** [comment]
Hi Glen, I think you and the team are doing a fantastic job.  I love unit tests / Travis. Is there anything small I can do to help get these tests merged?

**Comment by glenn-jocher** [comment]
@NauchtanRobotics thanks for the feedback! This particular PR is not meant for merging, we simply use it for evaluating support across the main Python releases.

If you see any room for CI improvements though please feel free to submit a PR. 

More broadly speaking one of our main challenges is CI on single and multi-GPU environments, which we don't do currently as we haven't found a great solution.

**Comment by NauchtanRobotics** [comment]
Thanks for the feedback that the CI script was intentionally remaining on a branch.

Regarding GPU testing, I guess Google colabs wouldn't be too excited about having freely offered resources consumed by automated testing (e.g  using colabctl: https://stackoverflow.com/questions/61762045/running-google-colab-every-day-at-a-specific-time)

**Comment by glenn-jocher** [comment]
@NauchtanRobotics yes, otherwise the CI times would be 4x their current times. It may make more sense to create two levels of CI, fast and default, where default would expand the testing across Python releases as in this branch and include more backend tests. 

We don't mind paying for a good solution, the important part is to ideally integrate with GitHub actions so that Single-GPU and Multi-GPU (i.e. ubuntu-latest with Python 3.8) would be 2 additional environments, i.e.:
https://github.com/ultralytics/yolov5/blob/8e5f9ddbdb3375d00483db1616ce891886072055/.github/workflows/ci-testing.yml#L18-L22

to
```yaml
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest, ubuntu-latest-gpu, ubuntu-latest-multigpu]
        python-version: [3.8]
        model: ['yolov5s']  # models to test

```

**Comment by glenn-jocher** [comment]
@Borda our convo above made me realize the coolest solution would be something like this that actually works:

> https://github.com/ultralytics/yolov5/blob/8e5f9ddbdb3375d00483db1616ce891886072055/.github/workflows/ci-testing.yml#L18-L22
> 
> to
> ```yaml
>       matrix:
>         os: [ubuntu-latest, macos-latest, windows-latest, ubuntu-latest-gpu, ubuntu-latest-multigpu]
>         python-version: [3.8]
>         model: ['yolov5s']  # models to test
> 
> ```

**Comment by Borda** [comment]
> our convo above made me realize the coolest solution would be something like this that actually works

that would be nice but the "multi-gpu" are not standard GH action agents so you would need to configure them somehow...
BUT, something in this direction you could do with Azure when you would be switching your agent pool

File: .github/workflows/ci-testing.yml [hunk 1]
```diff
@@ -19,7 +17,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        python-version: [ 3.9 ]
+        python-version: [ 3.6, 3.7, 3.8, 3.9 ]
```
Comment (Borda) [inline .github/workflows/ci-testing.yml:20]: 3.6 is ut of life cycle and I think it is fine just to test the oldest and latest python version, so 3.7 and 3.9
```suggestion
        python-version: [ 3.7, 3.9 ]
```

Comment (glenn-jocher) [reply .github/workflows/ci-testing.yml:20]: @Borda thanks for pointing this out! Do you think we should drop official support for 3.6 then? i.e. Update the README and the tutorials to require 3.7?

Screen Shot 2022-01-11 at 10 37 15 AM

Comment (Borda) [reply .github/workflows/ci-testing.yml:20]: as py3.6 is out of its life-cycle it is completely reasonable to drop it...

Comment (glenn-jocher) [reply .github/workflows/ci-testing.yml:20]: @Borda ok got it! I'll try to get that done. I think most of these changes are not in the code but in the tutorials etc. I'll have to coordinate the changes everywhere.

**Pull request closed without merging**